### PR TITLE
Support init of EarthRanger client from a token

### LINKED ISSUE #311: Support jwt initalisation in workflows EarthRanger client
Companion to https://github.com/wildlife-dynamics/ecoscope/issues/285

### PR DESCRIPTION
**Pull request by atmorling**
closes #311 

Depends on https://github.com/wildlife-dynamics/ecoscope/issues/285 being merged/released in order to bump the ecoscope version here

### CODE REVIEW

**Review by cisaacstern — COMMENTED** [review]
Thanks @atmorling! Looks quite close. A few comments:

- Let's also throw validation errors if:
    - All three of token _and_ username/password are provided
    - Username or password is provided as singletons (not a pair)
- I added the `test-io` label to this PR, which is required to get the tests for this to run (disabled it by default because it takes a long time and calls a 3rd party service). This should trigger io tests on the next commit pushed here.
- Added a small stylistic suggestion to your conditional logic in `get_client`. I think this way is a bit more readable to my eye, and generally nice to avoid conditional returns if possible (though the readability boost is moreso the motivation).
- I think we need to wait until ecoscope core library is released to get the necessary changes in, and then re-pin that release here, before merging, correct? (Wow, renaming `ecoscope-workflows` to something else, so we can stop saying "ecoscope core" will be kind of a relief, huh? 😅 )

File: src/ecoscope-workflows-ext-ecoscope/ecoscope_workflows_ext_ecoscope/connections.py [hunk 1]
```diff
@@ -41,22 +41,36 @@ class EarthRangerConnection(DataConnection[EarthRangerClientProtocol]):
...
+                sub_page_size=self.sub_page_size,
+            )
+        else:
+            return EarthRangerIO(
+                server=self.server,
+                username=self.username,
+                password=self.password.get_secret_value(),
+                tcp_limit=self.tcp_limit,
+                sub_page_size=self.sub_page_size,
+            )
```
[the remark this answers is not in the document]
Comment (atmorling) [reply src/ecoscope-workflows-ext-ecoscope/ecoscope_workflows_ext_ecoscope/connections.py]: Sure, done.

File: src/ecoscope-workflows-ext-ecoscope/ecoscope_workflows_ext_ecoscope/connections.py [hunk 2]
```diff
@@ -41,22 +41,36 @@ class EarthRangerConnection(DataConnection[EarthRangerClientProtocol]):
...
+    username: Annotated[str, Field(description="EarthRanger username")] = ""
+    password: Annotated[SecretStr, Field(description="EarthRanger password")] = ""
     tcp_limit: Annotated[int, Field(description="TCP limit for API requests")]
     sub_page_size: Annotated[int, Field(description="Sub page size for API requests")]
+    token: Annotated[SecretStr, Field(description="EarthRanger password")] = ""
+
+    @field_validator("token")
+    def token_or_password(cls, v: SecretStr, info: ValidationInfo):
+        if not v and not (info.data["username"] and info.data["password"]):
+            raise ValueError("EarthRanger username and password must be provided")
```
[the remark this answers is not in the document]
Comment (atmorling) [reply src/ecoscope-workflows-ext-ecoscope/ecoscope_workflows_ext_ecoscope/connections.py]: Done!

**Comment by atmorling** [comment]
Updated to cover your suggestions @cisaacstern 
Will get the ecoscope release/bump out tomorrow ideally

**Comment by cisaacstern** [comment]
I assume io test failures will resolve once ecoscope version is bumped. Though perhaps we will also need a new github secret (i.e. token)?

**Pull request merged**